src-django/api/field.py

```diff
@@ -14,3 +15,56 @@ def to_internal_value(self, data):
             raise serializers.ValidationError('All choices should be strings')
 
         return json.dumps(data)
+
+
+class ConditionTreeField(serializers.Field):
+    def to_representation(self, obj):
+        return json.loads(obj)
+
+    def to_internal_value(self, data):
+        if not isinstance(data, dict):
+            raise serializers.ValidationError('Expected dict')
+
+        self.__validate_json(data)
+
+        return json.dumps(data)
+
+    def __validate_json(self, data):
+        if 'node_type' not in data or not isinstance(data['node_type'], basestring):
+            raise serializers.ValidationError('Missing or invalid node type')
+
+        if data['node_type'] in ShowIf.CRITERIA_TYPES:
+            if 'children' in data:
+                raise serializers.ValidationError('CRITERIA node must have no children')
+
+            if 'criteria_element' not in data or not isinstance(data['criteria_element'], int):
+                raise serializers.ValidationError('CRITERIA must have an element')
+
+            try:
+                Element.objects.get(pk=data['criteria_element'])
+            except Element.DoesNotExist:
+                raise serializers.ValidationError('CRITERIA element is invalid')
+
+            if 'value' not in data or not isinstance(data['value'], basestring):
+                raise serializers.ValidationError('CRITERIA must have a value')
+        elif data['node_type'] in ShowIf.LOGICAL_TYPES:
+            if 'value' in data:
+                raise serializers.ValidationError('Only "CRITERIA" should have a value')
+
+            if 'criteria_element' in data:
+                raise serializers.ValidationError('Only "CRITERIA" should have an element')
+
+            if 'children' not in data or not isinstance(data['children'], list):
+                raise serializers.ValidationError('Logical nodes must have children')
+
+            if data['node_type'] == 'NOT' and len(data['children']) != 1:
+                raise serializers.ValidationError('NOT nodes must have exactly 1 child')
+
+            if data['node_type'] in ('AND', 'OR') and len(data['children']) < 2:
```

Comment:
AND and OR nodes should be allowed to have 1 child since it makes no logical difference to the user. The generator should unroll any logical nodes with only 1 child